lib/AST/Module.cpp

```diff
@@ -2685,6 +2685,38 @@ ModuleDecl::getDeclaringModuleAndBystander() {
   return *(declaringModuleAndBystander = {nullptr, Identifier()});
 }
 
+bool ModuleDecl::isClangOverlayOf(ModuleDecl *potentialUnderlying) {
+  return getUnderlyingModuleIfOverlay() == potentialUnderlying;
+}
+
+bool ModuleDecl::isSameModuleLookingThroughOverlays(
+  ModuleDecl *other) {
+  if (this == other) {
+    return true;
+  }
+
+  if (getUnderlyingModuleIfOverlay() == other) {
+    return true;
+  }
+
+  if (this == other->getUnderlyingModuleIfOverlay()) {
+    return true;
+  }
+  
+  // If the type has the @_originallyDefinedIn attribute, check if this is the
+  // "originally defined in" module.
+
+  if (getName().is(other->getAlternateModuleName())) {
+    return true;
+  }
+
+  if (other->getName().is(getAlternateModuleName())) {
+    return true;
+  }
```

Comment:
Wait—`this` and `other` are both modules, right? Is `getAlternateModuleName()` ever a thing for `ModuleDecl`s? If not, do you need to do something else to handle `@_originallyDefinedIn`?

Reply:
Yeah, so I just realized this check wasn't doing anything, but the existing check is sufficient to cover `@_originallyDefinedIn` (and I added a test to verify that)